tests/integration/api/controllers/place.spec.js

```diff
@@ -0,0 +1,80 @@
+const utils = require('@utils');
+const placeFactory = require('@factories/cht/contacts/place');
+const personFactory = require('@factories/cht/contacts/person');
+const { getRemoteDataContext, Place, Qualifier } = require('@medic/cht-datasource');
+const { expect } = require('chai');
+const userFactory = require('@factories/cht/users/users');
+
+describe('Place API', () => {
+  const contact0 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'chw' }));
+  const contact1 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'chw_supervisor' }));
+  const contact2 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'program_officer' }));
+  const placeMap = utils.deepFreeze(placeFactory.generateHierarchy());
+  const place1 = utils.deepFreeze({ ...placeMap.get('health_center'), contact: { _id: contact1._id } });
+  const place2 = utils.deepFreeze({ ...placeMap.get('district_hospital'), contact: { _id: contact2._id } });
+  const place0 = utils.deepFreeze({
+    ...placeMap.get('clinic'),
+    contact: { _id: contact0._id },
+    parent: {
+      _id: place1._id,
+      parent: {
+        _id: place2._id
+      }
+    },
+  });
+
+  const userNoPerms = utils.deepFreeze(userFactory.build({
+    place: place0._id,
+    roles: ['no_perms']
+  }));
+  const dataContext = getRemoteDataContext(utils.getOrigin());
+
+  before(async () => {
+    await utils.saveDocs([contact0, contact1, contact2, place0, place1, place2]);
+    await utils.createUsers([userNoPerms]);
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+    await utils.deleteUsers([userNoPerms]);
+  });
+
+  describe('GET /api/v1/place/:uuid', async () => {
+    const getPlace = Place.v1.get(dataContext);
+    const getPlaceWithLineage = Place.v1.getWithLineage(dataContext);
```

Comment:
In general, if you try to curry in an invalid dataContext (e.g. `null`, or some other value), this line will throw an exception.  

However, in this case, `dataContext` cannot be invalid here. The call above on L30 to `getRemoteDataContext` will either return a valid data context or it will throw an exception.

Reply:
We have a number of unit tests for these error scenarios in cht-datasource, but I did not think it was worth adding any integration tests since the validation logic all happens in cht-datasource and so there is not really anything to "integrate" with in those test cases.

A separate kind of test would be to create a remote data context with an invalid URL (one that does not actually point to a CHT instance). In that case, we would not fail until below when we call the `getPlaceWithLineage` function and that will throw an HTTP error (maybe a timeout error trying to connect to a server that does not exist??)  We don't have any test like that yet, (and probably does not belong in the place controller tests), but maybe we should consider some more generic cht-datasource integration tests... :thinking: Let me know if you want to see that as a part of this PR!